resolver.go

```diff
@@ -166,22 +172,36 @@ func setCommonFlags(msg *dns.Msg) {
 	msg.RecursionAvailable = true
 }
 
+func shuffleAddr(addr []net.IP) []net.IP {
+	for i := len(addr) - 1; i > 0; i-- {
```

Comment:
is there a reason to not iterate len(addr) times ? (`i > 0` implies we iterate it 1 less time)